symphony-ws/src/main/java/se/havochvatten/symphony/service/ReportService.java

```diff
@@ -48,25 +52,31 @@ public class ReportService {
     @Inject
     PropertiesService props;
 
-    static DoublePredicate not(DoublePredicate t) {
-        return t.negate();
-    }
-
-    private static DoubleStatistics getStatistics(Raster raster) {
-        DataBuffer buffer = raster.getDataBuffer();
-        DoubleStream stream = switch (buffer.getDataType()) {
-            case DataBuffer.TYPE_INT ->
-                Arrays.stream(((DataBufferInt)buffer).getData()).mapToDouble(x -> x);
-            case DataBuffer.TYPE_FLOAT ->
-                IntStream.range(0, ((DataBufferFloat)buffer).getData().length)
-                    .mapToDouble(i -> ((DataBufferFloat)buffer).getData()[i]);
-            case DataBuffer.TYPE_DOUBLE ->
-                Arrays.stream(((DataBufferDouble) buffer).getData());
-            default -> throw new RuntimeException("Unsupported raster data type");
-        };
-        // TODO report total as long to prevent overflow on big areas?
-        return stream.filter(not(CalcUtil.isNoData)).collect(DoubleStatistics::new,
-                DoubleStatistics::accept, DoubleStatistics::combine);
+    record StatisticsResult(double min, double max, double average, double stddev,
+                            int[] histogram, long zeroes, long pixels) { };
+
+    private static StatisticsResult getStatistics(GridCoverage2D coverage) {
+        RenderedImage image = coverage.getRenderedImage();
```

Comment:
One reason to use the CoverageProcessor or Operations-class is that you needn't unwrap the RenderedImage (and indeed you could not, since metadata like NODATA value and the like is stored in the GridCoverage and not the RenderedImage (IIUC).

Reply:
Looking at the implementation, the StatisticsOpImage constructor actually takes a simple Range object as a parameter to determine nodata values. 
I had an "example range" commented out earlier in development that I thought about leaving in as a template (for the event that we'll start using nodata - which may be soon enough) but ultimately opted not to. There's a helper method (I think I've seen it used elsewhere in Symphony) that converts the range to the datatype (although we'll probably use bytes).

I think there's another benefit to calling the underlying method directly, namely that you won't have to "buffer" the desired statistics operations by calling the operations one by one. Here, we just pass them all in.

Reply:
Not sure what you mean by buffer, but of course we would pass in the all statistics we are interested in to the operation (that's the "stats" parameter of the JAI-Ext "Stats" operation).

Reply:
This is what I mean:
https://github.com/havochvatten/MSP-Symphony/blob/8b76939a47066f825816dbb972a4b9237bd88e32/symphony-ws/src/main/java/se/havochvatten/symphony/calculation/CalibrationService.java#L88

Reply:
I see, that's obviously the corresponding parameter. I begin to see that my way of doing it seems pretty contrived.